article/_posts/2024-09-17-graphrag.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: review
+title: Retrieval Augmented Generation with Knowledge Graphs
+date: 2024-09-16
+time: 21:47
+reference: https://arxiv.org/abs/2404.16130
+tags:
+  - llm
+  - rag
+author: Jeremi Levesque
+cite:
+  authors: Edge, D., Trinh, H., Cheng, N., Bradley, J., Chao, A., Mody, A., ... & Larson, J.
+  title:From local to global: A graph rag approach to query-focused summarization.
+  venue: arXiv:2404.16130
+pdf: https://arxiv.org/abs/2404.16130
+summary: "Overview of the recent and popular GraphRAG method that utilizes a knowledge graph to retrieve information more relevant to the user's query. Utilizing a graph and community summaries, GraphRAG proposes a pipeline to answer questions that require a broader understanding of a dataset."
+---
+
+> **Quick Summary**
+> - GraphRAG is RAG where the retrieval path utilizes a knowledge graph.
+> - It improves the comprehensiveness and diversity of answers compared to RAG and it also enables questions/answers on more general and underlying themes of the data.
+> - Building the knowledge graph is the more complicated task than actually using GraphRAG.
+
+![](/article/images/graphrag/WebSearch-Evolution.png)
+*Figure 1: The evolution of Web Search. Google actually introduced the large scale use of a knowledge graph to enable the search engine to better understand what the user was searching for. Thus producing more relevant search results. In a way, GraphRAG is a combination of enabling a LLM to be fed relevant information from such a knowledge graph. Credits: This figure was taken from the presentation* [^4]
+# Graph RAG [^1]
+## Context
+At the time of writing, [knowledge graphs](https://en.wikipedia.org/wiki/Google_Knowledge_Graph) have been powering search across the web for around 12 years since [Google announced](https://blog.google/products/search/introducing-knowledge-graph-things-not/) in 2012 that the search experience would be enhanced by it. Knowledge graphs were meant to improve understanding of queries (especially some ambiguous queries) and to provide the user with a quick high-level summary of the things the user was possibly looking for. Also, it can be used to feed the user with possible new information that is linked in some way with the information on the subject he was searching for.
```

Comment:
```suggestion
At the time of writing, [knowledge graphs](https://en.wikipedia.org/wiki/Google_Knowledge_Graph) have been powering search across the web for around 12 years since [Google announced](https://blog.google/products/search/introducing-knowledge-graph-things-not/) in 2012 that the search experience would be enhanced by them. Knowledge graphs were meant to improve understanding of queries (especially some ambiguous queries) and to provide the user with a quick high-level summary of the things the user was possibly looking for. Also, it can be used to feed the user with possible new information that is linked in some way with the information on the subject he was searching for.
```